Add optional type field for MIME types

ninbryan opened the pull request:
optional script type

ninbryan committed:
Update basket.js

optional script type

addyosmani commented:
Please include a description with your PR. Your comment mentions JSX in the diff but you don't explain why this would be useful to other users :)

felipemorais commented:
Mentions about JSX aren't relevant.
+1 to custom type attribute.

ninbryan committed:
Update basket.js

ninbryan committed:
add optional type field

optional type property for details in basket.require(details)
MIME type field allows other languages including dialects of JavaScript to compile

ninbryan commented:
I was looking into how to word this better.
Sorry for the delay.
